File: app/frontend/src/utils/realm/schema/Hall.ts
import {Realm} from '@realm/react';

class Hall extends Realm.Object<Hall> {
  hall_id!: string;
  name!: string;
  address!: string;
  static schema: Realm.ObjectSchema = {
    name: 'Hall',
    properties: {
      hall_id: 'string',
      name: 'string',
      address: 'string',
    },
    primaryKey: 'hall_id',
  };
}

export default Hall;

File: server/batch-service/src/main/java/com/c209/batchservice/domain/catalog/entity/Hall.java
/**
 * This code was COPIED from "com.c209.catalog" package.
 */
package com.c209.batchservice.domain.catalog.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.Getter;

import static jakarta.persistence.GenerationType.IDENTITY;

@Getter
@Entity
public class Hall {
    @GeneratedValue(strategy = IDENTITY)
    @Column(name = "hall_id")
    @Id
    private Long id;
    private String name;
    private String address;
}
